### ISSUE TITLE
Don't ignore empty array in "whereIn" method

### ISSUE THREAD

**Issue by AeonFr** [ClanCats/Hydrahon#21]
I've noticed the `whereIn` method doesn't apply the "where" clause when the array is empty.

``` php
    public function whereIn($column, array $options = array())
    {
        // when the options are empty we skip
        if ( empty( $options ) )
        {
            return $this;
        }

        return $this->where($column, 'in', $options);
    }
```

I think this is kind of obscure behavior.

The current implementation of the method seems like doing this:

``` php
if (count($items) == 0) {
    $query->execute();
} else {
   $query->whereIn('item', $items)->execute();
}
```

I think the case scenario where you would want to do this is an exception rather than the norm, meaning I would not like it as a default behavior.

**What happened in my project...**

I thought of giving a real-life example to illustrate why this might be undesirable.

I'm working in a project where user is authenticated in one or more offices.
I ran a `whereIn` query to see if user_id appears in "offices" table.

When the user doesn't appear in any table, instead of getting a MySQL error, I get as a return all of the offices.

If the user is not well configured, I would rather see him having an empty screen (`where in ()` is invalid SQL, so it would crash the app), rather than giving him access to all offices!

**In conclusion**

Dont' make `whereIn` ignore the second argument if an empty array is passed.

I can send a PR since I've already implemented this change in [a commit of my fork of the repo](https://github.com/AeonFr/Hydrahon/commit/2cf21b09aae2b20ae9a0d5bf04db3584ae256177).

Thanks for reading

**Edit:**

Another option would be that, if `whereIn` has an empty array, make it the SQL equivalent of `where false`.

I think that's a sensible default and wouldn't cause a MySQL error. After all, it's technically true that there should be *no* records if you pass an empty argument in such a query.

**Comment by mario-deluna** [comment]
I completely agree, this should throw an exception. If the second argument is empty `whereIn` should not be called in the first place. 

As for now I do not want to implement this change on `v1` because it might break current  implementations that expect empty `whereIn` statements to be ignored. 

This might be the case in something like this:

```php
$categories = explode(',', $payload->query->get('categories'));

$h->table('posts')
    ->select()
    ->whereIn('category', $categories)
    ->get();
```

This is something that definitely should be fixed for `v2` (https://github.com/ClanCats/Hydrahon/tree/v2)

**Comment by AeonFr** [comment]
Ok, considering it involves a breaking change I will close the issue. 

It's good to know its considered for next version. 

Thanks
